Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case 3 - DRAG calibration/rb.py

```diff
@@ -0,0 +1,259 @@
+                state, I, Q = readout_macro(threshold=ge_threshold, state=state, I=I, Q=Q)
+                # discriminator.measure_state("readout", "out1", "out2", res=state, I=I, Q=Q)
+
+                save(state, state_st)
+                save(Q, Q_st)
+                save(I, I_st)
+
+                assign(sequence_list[depth], saved_gate)
+
+    with stream_processing():
+        n_st.save("iteration")
+        state_st.boolean_to_int().buffer(n_avgs).map(FUNCTIONS.average()).buffer(
+            num_of_sequences, max_circuit_depth
+        ).save("res")
+        Q_st.buffer(n_avgs).map(FUNCTIONS.average()).buffer(num_of_sequences, max_circuit_depth).save("Q")
+        I_st.buffer(n_avgs).map(FUNCTIONS.average()).buffer(num_of_sequences, max_circuit_depth).save("I")
+
+qm = qmm.open_qm(config, close_other_machines=False)
+
+job = qm.execute(rb)
+res_handles = job.result_handles
+# res_handles.wait_for_all_values()
+iteration_handle = res_handles.get("iteration")
+iteration_handle.wait_for_values(1)
+next_percent = 0.1  # First time print 10%
+
+print("Progress =", end=" ")
+
+while res_handles.is_processing():
+
+    iteration = iteration_handle.fetch_all()
+    if iteration / num_of_sequences > next_percent:
+        percent = 10 * round(iteration / num_of_sequences * 10)  # Round to nearest 10%
+        print(f"{percent}%", end=" ")
+        next_percent = percent / 100 + 0.1  # Print every 10%
+
+resvalue = res_handles.res.fetch_all()
+Q = res_handles.get("Q").fetch_all()
+I = res_handles.get("I").fetch_all()
+
+value = 1 - np.average(resvalue, axis=0)
+Q = np.average(Q, axis=0)
+I = np.average(I, axis=0)
+
+plt.plot(value)
+
+# plt.figure()
+# plt.plot(Q)
+
+# plt.figure()
+# plt.plot(I)
+
+
+def power_law(m, a, b, p):
+    return a * (p**m) + b
+
+
+x = np.linspace(1, max_circuit_depth, max_circuit_depth)
+plt.xlabel("Number of cliffords")
+plt.ylabel("Sequence Fidelity")
+
+pars, cov = curve_fit(
+    f=power_law,
+    xdata=x,
+    ydata=value,
+    p0=[0.5, 0.5, 0.9],
+    bounds=(-np.inf, np.inf),
+    maxfev=2000,
+)
+
+plt.plot(x, power_law(x, *pars), linestyle="--", linewidth=2)
+
+stdevs = np.sqrt(np.diag(cov))
+
+print("#########################")
+print("### Fitted Parameters ###")
+print("#########################")
+print(f"A = {pars[0]:.3} ({stdevs[0]:.1}), B = {pars[1]:.3} ({stdevs[1]:.1}), p = {pars[2]:.3} ({stdevs[2]:.1})")
+print("Covariance Matrix")
+print(cov)
+
+one_minus_p = 1 - pars[2]
+r_c = one_minus_p * (1 - 1 / 2**1)
+r_g = r_c / 1.875
+r_c_std = stdevs[2] * (1 - 1 / 2**1)
+r_g_std = r_c_std / 1.875
+
+print("#########################")
+print("### Useful Parameters ###")
+print("#########################")
+print(
+    f"1-p = {np.format_float_scientific(one_minus_p, precision=2)} ({stdevs[2]:.1}), "
+    f"r_c = {np.format_float_scientific(r_c, precision=2)} ({r_c_std:.1}), "
+    f"r_g = {np.format_float_scientific(r_g, precision=2)}  ({r_g_std:.1})"
+)
```

Comment:
The newer RB code has an explanation on these parameters, maybe it's better to copy these lines here as well?

Reply:
thanks